src/main.rs

```diff
@@ -254,71 +251,73 @@ struct Args {
     ██╔██╗ ██║███████║  ███╔╝ ███████║██████╔╝███████║
     ██║╚██╗██║██╔══██║ ███╔╝  ██╔══██║██╔══██╗██╔══██║
     ██║ ╚████║██║  ██║███████╗██║  ██║██║  ██║██║  ██║
     ╚═╝  ╚═══╝╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝╚═╝  ╚═╝╚═╝  ╚═╝
+    (c) Tiara Hock aka ByteOtter. (github.com/ByteOtter)
+
+    Licensed under the terms of the GPL-v3.0 License.
+    Check github.com/The-Nazara-Project/Nazara/LICENSE for more info.
 "#;
 
     // Welcome Message.
-    println!(
-        "{} \n(c) Tiara Hock aka ByteOtter. (github.com/ByteOtter)\n
-         Licensed under the terms of the GPL-v3.0 License.\n\
-         Check github.com/The-Nazara-Project/Nazara/LICENSE for more info.\n",
-        ascii_art
-    );
-
-    let config = match set_up_configuration(args.uri, args.token, args.name.clone()) {
-        Ok(conf) => conf,
-        Err(err) => {
-            err.abort(None);
-        }
-    };
-
-    let client: ThanixClient = ThanixClient {
-        base_url: config.get_netbox_uri().to_string(),
-        authentication_token: config.get_api_token().to_string(),
-        client: Client::new(),
-    };
+    println!("{}", ASCII_ART);
 
-    match probe(&client) {
-        Ok(()) => {}
-        Err(err) => err.abort(None),
-    };
-
-    let dmi_information: dmi_collector::DmiInformation =
-        dmi_collector::construct_dmi_information()?;
-
-    let network_information: Vec<NetworkInformation> =
-        network_collector::construct_network_information()?;
-
-    let machine: Machine = Machine {
-        name: args.name,
-        dmi_information,
-        network_information,
+    // Collect machine information.
+    let machine = Machine {
+        name: args.name.clone(),
+        dmi_information: dmi_collector::construct_dmi_information()?,
+        network_information: network_collector::construct_network_information()?,
         custom_information: match execute(args.plugin) {
             Ok(info) => Some(info),
             Err(e) => panic!("{}", e.to_string()),
         },
     };
 
-    // Passing a name in any way is mandatory for a virtual machine
+    // Passing a name in any way is mandatory for a virtual machine.
     if machine.dmi_information.system_information.is_virtual && machine.name.is_none() {
         eprintln!(
             "[FATAL] No name has been provided for this virtual machine! Providing a name as search parameter is mandatory for virtual machines."
         );
         process::exit(1)
     }
 
-    // Register the machine or VM with NetBox
-    match register_machine(&client, machine, config) {
-        Ok(_) => {
-            println!("\x1b[32mAll done, have a nice day!\x1b[0m");
-            process::exit(0);
+    // If we only want to do a dry run, we only have to print the collected information.
+    if args.dry_run {
+        println!("Dry run results:");
+        dbg!(&machine);
+    } else {
+        let config = match set_up_configuration(args.uri, args.token, args.name.clone()) {
```

Comment:
Maybe we could pull the parsing of the config file out so we can also print the `config` and `client` structs?

That would bring a complete overview over all data that Nazara has collected.